### PR TITLE
#166705699 - Get an error for the unavailable URL

### PR DESCRIPTION
**Pull request by UhiriweAudace**
#### What does this PR do?
In this PR, I added a middleware that will track if the URL provided is available or reachable
#### Description of Task to be completed?
having the following endpoint, we return an error message to the user:
`http://localhost/fdfasfasfafasfaasffaf`
`http://localhost/api/fdfasfasfafasfaasffaf`
#### How should this be manually tested?
- clone the repository
- go to your terminal and then write the following command
  `cd tesla-ah`
- after that try to install all dependencies that are required for this project, simply by running the following command:
   `yarn install`  or `yarn`
- Then running the server, by running the below commands:
   1. `sequelize db:migrate` for this command make sure that the `sequelize-cli` package is installed globally
   2. `yarn dev`
   3.  After running the server, Open your postman, and then try to send a request with a bad URL
        you will get an error message
- for running the tests, write the below command in your terminal
     `yarn test`
 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166705699 ](https://www.pivotaltracker.com/n/projects/2350798/stories/166705699)
#### Screenshots (if appropriate)
N/A
#### Questions:

****
**Before Code - Audace Uhiriwe**
https://docs.google.com/document/d/1mEyswr6qQIkYr93ekXAXCK39t8toLdhu8NQ94n2ahXw/edit?usp=sharing

### CODE REVIEW

**Review by eliemugenzi — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -15,9 +15,14 @@ const port = process.env.PORT || 3000;
...
-app.use('/api', api);
-app.use('/', swaggerUi.serve, swaggerUi.setup(swaggerDoc));
 
+app.use('/api', api);
+app.use('*', (req, res) => {
+  res.status(404).send({
+    error: 'This URL Not found'
+  });
+});
+app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerDoc));
```
Comment (mifeille) [inline src/index.js]: why did you add /"docs" on the swagger route?

File: test/article.js [hunk 2]
```diff
@@ -17,6 +17,19 @@ chai.use(chaiHttp);
 
 let validToken, slug, unverifiedToken;
 let updatedSlug, invalidToken;
+
+describe('Unavailable URL', () => {
+  it('Should return an error message, Once you tried to access the unavaliable URL', () => {
```
Comment (deschantkn) [inline test/article.js]: this test does not belong in this file. Instead you should create a `server.test.js` file for it.

**Review by deschantkn — DISMISSED** [review]

**Review by rukundoeric — APPROVED** [review]

File: test/article.js (hunk 2 above)
Comment (UhiriweAudace) [reply test/article.js]: that is a good idea @deschantkn

**Review by EmyRukundo — APPROVED** [review]

File: src/index.js [hunk 3]
```diff
@@ -24,11 +24,16 @@ app.use(session({
...
-app.use('/api', api);
-app.use('/', swaggerUi.serve, swaggerUi.setup(swaggerDoc));
 app.use(passport.initialize());
 app.use(passport.session());
 
+app.use('/api', api);
+app.use('/docs', swaggerUi.serve, swaggerUi.setup(swaggerDoc));
+app.use('*', (req, res) => {
+  res.status(404).send({
+    error: 'This URL Not found'
```
Comment (Quantum-35) [inline src/index.js]: Kindly update your Error message from `URL not Found to` Page not found.

**Review by Quantum-35 — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 3 above)
Comment (UhiriweAudace) [reply src/index.js]: okay

**Review by Quantum-35 — APPROVED** [review]
